[receiver/datadog]Flaky test TestDefaultReceivers/datadog/lifecycle

Seen in https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5249506268/jobs/9483727674?pr=23332

```
%v(0xc9a65a0,0x10ddf780)%v(0xc9a65a0,0x10ddf780)--- FAIL: TestDefaultReceivers (60.93s)
    --- FAIL: TestDefaultReceivers/datadog (0.00s)
        --- FAIL: TestDefaultReceivers/datadog/lifecycle (0.00s)
            receivers_test.go:[52](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5249506268/jobs/9483727674?pr=23332#step:9:53)3: 
                	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/otelcontribcol/receivers_test.go:523
                	            				/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/otelcontribcol/receivers_test.go:481
                	Error:      	Received unexpected error:
                	            	failed to create datadog listener: listen tcp 127.0.0.1:8126: bind: address already in use
                	Test:       	TestDefaultReceivers/datadog/lifecycle
FAIL
	github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontribcol	coverage: 44.8% of statements
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontribcol	229.4[59](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/5249506268/jobs/9483727674?pr=23332#step:9:60)s
FAIL
make[2]: *** [../../Makefile.Common:103: test-with-cover] Error 1
make[1]: *** [Makefile:172: cmd/otelcontribcol] Error 2
make[2]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/otelcontribcol'
make: *** [Makefile:100: gotest-with-cover] Error 2
make[1]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib'
Error: Process completed with exit code 2.
```

--- a/cmd/otelcontribcol/receivers_test.go
+++ b/cmd/otelcontribcol/receivers_test.go
@@ -160,7 +160,7 @@ func TestDefaultReceivers(t *testing.T) {
 			receiver: "datadog",
 			getConfigFn: func() component.Config {
 				cfg := rcvrFactories["datadog"].CreateDefaultConfig().(*datadogreceiver.Config)
-				cfg.Endpoint = ":0" // Using a randomly assigned address
+				cfg.Endpoint = "localhost:0" // Using a randomly assigned address
 				return cfg
 			},
 		},

--- a/cmd/otelcontribcol/receivers_test.go
+++ b/cmd/otelcontribcol/receivers_test.go
@@ -29,6 +29,7 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/azuremonitorreceiver"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/carbonreceiver"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/chronyreceiver"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/datadogreceiver"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/filelogreceiver"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/jmxreceiver"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbatlasreceiver"
@@ -157,6 +158,11 @@ func TestDefaultReceivers(t *testing.T) {
 		},
 		{
 			receiver: "datadog",
+			getConfigFn: func() component.Config {
+				cfg := rcvrFactories["datadog"].CreateDefaultConfig().(*datadogreceiver.Config)
+				cfg.Endpoint = ":0" // Using a randomly assigned address
+				return cfg
+			},
 		},
 		{
 			receiver:     "docker_stats",

